src/main/java/seedu/address/logic/commands/RemoveGradeCommand.java

```diff
@@ -0,0 +1,112 @@
+import seedu.address.model.Model;
+import seedu.address.model.assignment.Assignment;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Github;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Telegram;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Remove assignment grades from an existing person in the address book.
+ */
+public class RemoveGradeCommand extends Command {
+    public static final String COMMAND_WORD = "removeGrade";
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD
+                    + ": Removes a grade of an assignment from the person.\n"
+                    + "Parameters: "
+                    + PREFIX_NAME
+                    + "NAME "
+                    + PREFIX_ASSIGNMENT
+                    + "ASSIGNMENT "
+                    + "Example: "
+                    + COMMAND_WORD
+                    + " "
+                    + PREFIX_NAME
+                    + "John Doe "
+                    + PREFIX_ASSIGNMENT
+                    + "Ex09 ";
+
+    public static final String MESSAGE_SUCCESS = "Assignment %1$s removed from %2$s";
+    public static final String MESSAGE_FAILURE = "Assignment %s already removed for %s.";
+    private final Name personName;
+    private final String assignmentName;
+
+    /**
+     * @param personName     Name of the person.
+     * @param assignmentName Name of assignment.
+     */
+    public RemoveGradeCommand(String personName, String assignmentName) {
+        requireAllNonNull(personName, assignmentName);
+        this.personName = new Name(personName);
+        this.assignmentName = assignmentName;
+    }
+
+
+    static Person createPersonWithRemovedGrade(Person person, String assignmentName) {
+        assert person != null;
+        Name name = person.getName();
+        Phone phone = person.getPhone();
+        Email email = person.getEmail();
+        Set<Tag> tags = person.getTags();
+        Telegram telegram = person.getTelegram();
+        Github github = person.getGithub();
+        Set<Integer> weeksAttended = person.getWeeksPresent();
+
+        Map<String, Assignment> assignment = person.getAssignment();
+        assignment.remove(assignmentName);
+
+        return new Person(name, phone, email, telegram, github, assignment, weeksAttended, tags);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // check if assignment is in predefined list
+        if (!model.hasAssignment(assignmentName)) {
+            throw new CommandException("Invalid assignment name: " + assignmentName);
+        }
+
+        Person person = model.getPerson(personName)
+                .orElseThrow(() ->
+                        new CommandException("Person " + personName + " not in address book"));
```

Comment:
I think under `Messages.MESSAGE_INVALID_PERSON_DISPLAYED_NAME` there is an error message similar this.